### DISCUSSION TITLE
How to access query from an uncontrolled Builder

### DISCUSSION THREAD

**Discussion by trebitowski** [react-querybuilder/react-querybuilder#666] [General]
I'm trying to use an uncontrolled QueryBuilder (uses defaultQuery prop), and access the query when submitting a form. However the QueryBuilder component doesn't accept a ref/have any imperative controls. Is there a way to get a query from an uncontrolled QueryBuilder?

**Comment by jakeboone02** [comment]
The [`onQueryChange`](https://react-querybuilder.js.org/docs/components/querybuilder#onquerychange) callback prop will fire every time the query is updated (and once on mount), even if the component is uncontrolled.

Version 7 will have a couple of other ways to keep track of the query, but that should be enough for any previous versions.
